src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -37,226 +37,283 @@
  */
 public class Parser {
 
-    /**
-     * Used for initial separation of command word and args.
-     */
-    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+	/**
+	 * Used for initial separation of command word and args.
+	 */
```

Comment:
oh i got it to work, will do the checkstyle changes, but first ill finish make task abstract